asciidoc/cypher-workflow.adoc

```diff
@@ -0,0 +1,804 @@
+This avoids the small latency overhead of the causal chain.
+
+Bookmarks can be passed between sessions by extracting the last bookmark from a session and passing this into the construction of another.
+Multiple bookmarks can also be combined if a transaction has more than one logical predecessor.
+Note that it is only when chaining across sessions that an application will need to work with bookmarks directly.
+
+
+[[driver-access-modes-transaction]]
+== Routing transactions using access modes
+
+Transactions can be executed in either read or write mode; this is known as the access mode.
+In a Causal Cluster, each transaction will be routed to an appropriate server based on the mode.
+When using a single instance, all transactions will be passed to that one server.
+
+Routing Cypher by identifying reads and writes can improve the utilization of available cluster resources.
+Since read servers are typically more plentiful than write servers, it is beneficial to direct read traffic to read servers instead of the write server.
+Doing so helps in keeping write servers available for write transactions.
+
+Access mode is generally specified by the method used to call the transaction function.
+Session classes provide a method for calling reads and another for writes.
+
+As a fallback for auto-commit and unmanaged transactions, a default access mode can also be provided at session level.
+This is only used in cases when the access mode cannot otherwise be specified, and will be overridden should a transaction function be used within that session.
+
+[NOTE]
+The driver does not parse Cypher and therefore cannot automatically determine whether a transaction is intended to carry out read or write operations.
+As a result, a write transaction tagged as a read will still be sent to a read server, but will fail on execution.
+
+[.tabbed-example]
+.Read-write transaction
+====
+[.include-with-dotnet]
+======
+[source, csharp]
+----
+include::{dotnet-examples}/Examples.cs[tags=read-write-transaction]
+----
+======
+
+// [.include-with-go]
+// ======
+// [source, go]
+// ----
+// include::{go-examples}/examples_test.go[tags=read-write-transaction]
+// ----
+// ======
+
+[.include-with-java]
+======
+[source, java]
+----
+include::{java-examples}/ReadWriteTransactionExample.java[tags=read-write-transaction]
+----
+======
+
+[.include-with-javascript]
+======
+[source, javascript]
+----
+include::{javascript-examples}/examples.test.js[tags=read-write-transaction]
+----
+======
+
+// [.include-with-python]
+// ======
+// [source, python]
+// ----
+// include::{python-examples}/read_write_transaction_example.py[tags=read-write-transaction]
+// ----
+// ======
+====
+
+[[driver-databases]]
+== Databases and execution context
+
+Neo4j 4.0 introduces the ability to work with multiple databases within the same DBMS.
+For a single instance, this is limited to one user database, plus the `system` database.
```

Comment:
I think this sentence is obsolete here.

Reply:
I was attempting to highlight the difference in features/surface between the Enterprise and Community products. There may be a better way to do this.

Reply:
Left as is, it serves Nigel's purpose.